nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -9,490 +9,492 @@
+    "test_shape, test_affine, msg",
     [
-        ((2, 3, 5), np.eye(4), _TEST_SHAPE_ERROR_MSG),
-        ((2, 3, 4), 2 * np.eye(4), _TEST_AFFINE_ERROR_MSG),
+        ((8, 9, 11), AFFINE, _TEST_SHAPE_ERROR_MSG),
+        (SHAPE, 2 * AFFINE, _TEST_AFFINE_ERROR_MSG),
     ],
 )
-def test__check_shape_and_affine_compatibility_error(
-    test_shape, test_affine, msg
+def test_check_shape_and_affine_compatibility_error(
+    img_3D, test_shape, test_affine, msg
 ):
-    shape = (2, 3, 4)
-    affine = np.eye(4)
-    img1 = nibabel.Nifti1Image(np.zeros(shape), affine)
-
     img2 = nibabel.Nifti1Image(np.zeros(test_shape), test_affine)
 
     with pytest.raises(ValueError, match=msg):
-        signal_extraction._check_shape_and_affine_compatibility(img1, img2)
-
-
-def test_generate_regions_ts():
-    """Minimal testing of generate_regions_ts()."""
-    # Check that no regions overlap
-    n_voxels = 50
-    n_regions = 10
-    regions = generate_regions_ts(n_voxels, n_regions, overlap=0)
-    assert regions.shape == (n_regions, n_voxels)
-    # check: no overlap
-    np.testing.assert_array_less(
-        (regions > 0).sum(axis=0) - 0.1, np.ones(regions.shape[1])
-    )
-    # check: a region everywhere
-    np.testing.assert_array_less(
-        np.zeros(regions.shape[1]), (regions > 0).sum(axis=0)
-    )
+        _check_shape_and_affine_compatibility(img1=img_3D, img2=img2)
 
-    regions = generate_regions_ts(
-        n_voxels, n_regions, overlap=0, window="hamming"
-    )
-    assert regions.shape == (n_regions, n_voxels)
-    # check: no overlap
-    np.testing.assert_array_less(
-        (regions > 0).sum(axis=0) - 0.1, np.ones(regions.shape[1])
-    )
-    # check: a region everywhere
-    np.testing.assert_array_less(
-        np.zeros(regions.shape[1]), (regions > 0).sum(axis=0)
-    )
 
-    # Check that some regions overlap
-    regions = generate_regions_ts(n_voxels, n_regions, overlap=1)
-    assert regions.shape == (n_regions, n_voxels)
-    assert np.any((regions > 0).sum(axis=-1) > 1.9)
+def test_errors_3D(img_3D):
+    """Verify that 3D images are refused."""
+    wrong_dim_image = img_3D
 
-    regions = generate_regions_ts(
-        n_voxels, n_regions, overlap=1, window="hamming"
-    )
-    assert np.any((regions > 0).sum(axis=-1) > 1.9)
+    with pytest.raises(DimensionError, match=_4D_EXPECTED_ERROR_MSG):
+        img_to_signals_labels(imgs=wrong_dim_image, labels_img=img_3D)
 
+    with pytest.raises(DimensionError, match=_4D_EXPECTED_ERROR_MSG):
+        img_to_signals_maps(imgs=img_4D, maps_img=wrong_dim_image)
 
-def test_generate_labeled_regions():
-    """Minimal testing of generate_labeled_regions."""
-    shape = (3, 4, 5)
-    n_regions = 10
-    regions = generate_labeled_regions(shape, n_regions)
-    assert regions.shape == shape
-    assert len(np.unique(get_data(regions))) == n_regions + 1
 
+def test_errors_4D(img_3D, img_4D):
+    """Verify that 4D images are refused."""
+    # 4D labels
```

Comment:
I'd remove this comment and split the test: `test_errors_4D_labels` and `test_errors_4D_mask`

Reply:
done